lib/proxyHandlers.js

```diff
@@ -0,0 +1,13 @@
+const util = require('util');
+
+const proxyHandler = {
+  get: function (obj, prop) {
+    switch (typeof obj[prop]) {
+      case 'function': return util.promisify(obj[prop]);
+      case 'object': return new Proxy(obj[prop], proxyHandler);
+      default: return obj[prop];
+    }
+  },
+};
+
+module.exports = proxyHandler;
```

Comment:
What was your thinking for this being a separate file to `asyncBot.js`?

Reply:
@tonytamps just thinking we might use `proxyHandler` for other stuff, cause it looks quite general.

Reply:
In my mind, this is what makes up asyncBot. 
It feels like you're doing a couple things here. 

Firstly, what I call "prefactoring" where you attempt to refactor before there is any need to.

Second thing I think you're doing is seeing proxys as something that you group around (like controllers or services). Proxy is an implementation detail. It's no different to a `class`, `function`, `boolean`. You don't tend to group around these language constructs.

I would merge files together in `asyncBot.js`.


Feel free to disregard what I would do though. It's just my opinion. 😄

Reply:
@tonytamps yeah that's fair enough, I also think that would make code more consistent and easy to read. Have merged them in the latest commit.